packages/components/combobox/src/combobox.ts

```diff
@@ -0,0 +1,603 @@
+      >
+        ${this.multiple && this.currentSelection.length
+          ? html`
+              <sl-tag-list
+                slot="prefix"
+                size=${ifDefined(this.size)}
+                stacked
+                .emphasis=${this.disabled ? 'bold' : 'subtle'}
+              >
+                ${this.currentSelection.map(
+                  option => html`
+                    <sl-tag
+                      @sl-remove=${() => this.#onRemove(option)}
+                      ?disabled=${this.disabled}
+                      ?removable=${!this.disabled}
+                    >
+                      ${option.content}
+                    </sl-tag>
+                  `
+                )}
+              </sl-tag-list>
+            `
+          : nothing}
+        <slot name="input" slot="input"></slot>
+        <button
+          @click=${this.#onButtonClick}
+          ?disabled=${this.disabled}
+          aria-label=${msg(str`${this.listbox?.matches(':popover-open') ? 'Hide' : 'Show'} the options`)}
+          slot="suffix"
+          tabindex="-1"
+        >
+          <sl-icon name="chevron-down"></sl-icon>
+        </button>
+      </sl-text-field>
+
+      <slot
+        ${anchor({
+          element: this,
+          offset: Combobox.offset,
+          position: 'bottom-start',
+          viewportMargin: Combobox.viewportMargin
+        })}
+        @beforetoggle=${this.#onBeforeToggle}
+        @click=${this.#onOptionsClick}
+        @slotchange=${() => this.#updateOptions()}
+        @toggle=${this.#onToggle}
+        part="wrapper"
+        popover
+        tabindex="-1"
+      ></slot>
+    `;
+  }
+
+  #onBeforeToggle(event: ToggleEvent): void {
+    this.input.setAttribute('aria-expanded', event.newState === 'open' ? 'true' : 'false');
+
+    if (event.newState === 'open') {
+      this.wrapper!.style.inlineSize = `${this.getBoundingClientRect().width}px`;
+    } else {
+      this.#popoverJustClosed = true;
+    }
+  }
+
+  #onButtonClick(): void {
+    if (!this.#popoverJustClosed) {
+      this.wrapper?.showPopover();
+    }
+  }
+
+  #onFocus(): void {
+    if (!this.#pointerDown) {
+      this.wrapper?.showPopover();
+    }
+  }
+
+  #onInput(event: InputEvent): void {
+    const value = this.input.value;
+
+    this.wrapper?.showPopover();
+
+    let currentOption: ComboboxOption | undefined = undefined;
+    if (
+      event.inputType !== 'deleteContentBackward' &&
+      (this.autocomplete === 'inline' || this.autocomplete === 'both')
+    ) {
+      currentOption = this.options.find(option => option.content.toLowerCase().startsWith(value.toLowerCase()));
+
+      if (currentOption) {
+        this.input.value = currentOption.content;
+        this.input.setSelectionRange(value.length, currentOption.content.length);
+      }
+    } else {
+      currentOption = this.options.find(option => value === option.value);
+    }
+
+    this.#updateCurrent(currentOption);
+    this.#updateFilteredOptions(value);
+
```

Comment:
Maybe after selecting an option we should clean the input?

<img width="852" alt="image" src="https://github.com/user-attachments/assets/2b1ab259-af49-459d-8a63-555e7618cd05">

Reply:
Done